Eastern-Research-Group/mywaterway
# Feature/452 separate surrounding visibility

## Related Issues:
* [HMW-452](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=864&view=detail&selectedIssue=HMW-452)

## Main Changes:
* Pulled the HUC and surrounding features layers out from their parent Group layers.
* Stopped linking surrounding feature layer visibility with HUC layer visibility.
  * Additionally, surrounding layers maintain their visibility between tabs, rather than match the visibility of their corresponding layers in the layer list.
* Switched to creating the `layers` object passed to the map from the layers in the Layers context. This was done to address an issue with surrounding layers when switching to a fullscreen map on the _Community_ page, and to further ensure that the `layers` object is always up to date.

## Steps To Test:
1. Test toggling on various layers in the layer list and surrounding features widget, and confirm they operate independently.
2. On the _Community_ page, toggle on a surrounding features layer, then switch tabs. Confirm the layer is still visible.
3. On all pages, toggle on a surrounding features layer, then activate the fullscreen map widget. Confirm the surrounding layer still updates as the map is panned.
4. Save the map with the "Add & Save Data Widget", and confirm that surrounding feature layers (except for the `allWaterbodiesLayer`) are excluded from the hosted feature service and web map.
